x-pack/plugins/observability_solution/observability_ai_assistant_app/server/plugin.ts

```diff
@@ -42,6 +54,68 @@ export class ObservabilityAIAssistantAppPlugin
     >,
     plugins: ObservabilityAIAssistantAppPluginSetupDependencies
   ): ObservabilityAIAssistantAppServerSetup {
+    const routeHandlerPlugins = mapValues(plugins, (value, key) => {
+      return {
+        setup: value,
+        start: () =>
+          core.getStartServices().then((services) => {
+            const [, pluginsStartContracts] = services;
+            return pluginsStartContracts[
+              key as keyof ObservabilityAIAssistantPluginStartDependencies
+            ];
+          }),
+      };
+    }) as ObservabilityAIAssistantRouteHandlerResources['plugins'];
+
+    const initResources = async (
+      request: KibanaRequest
+    ): Promise<ObservabilityAIAssistantRouteHandlerResources> => {
+      const [coreStart, pluginsStart] = await core.getStartServices();
+      const license = await firstValueFrom(pluginsStart.licensing.license$);
+      const savedObjectsClient = coreStart.savedObjects.getScopedClient(request);
+
+      const context: ObservabilityAIAssistantRequestHandlerContext = {
+        rac: routeHandlerPlugins.ruleRegistry.start().then((startContract) => {
+          return {
+            getAlertsClient() {
+              return startContract.getRacClientWithRequest(request);
+            },
+          };
+        }),
+        alerting: routeHandlerPlugins.alerting.start().then((startContract) => {
+          return {
+            getRulesClient() {
+              return startContract.getRulesClientWithRequest(request);
+            },
+          };
+        }),
+        core: Promise.resolve({
+          coreStart,
+          elasticsearch: {
+            client: coreStart.elasticsearch.client.asScoped(request),
+          },
+          uiSettings: {
+            client: coreStart.uiSettings.asScopedToClient(savedObjectsClient),
+          },
+          savedObjects: {
+            client: savedObjectsClient,
+          },
+        }),
+        licensing: Promise.resolve({ license, featureUsage: pluginsStart.licensing.featureUsage }),
```

Comment:
Do we have to wrap these values as promises? Could we change the type in `ObservabilityAIAssistantRequestHandlerContext`?

Reply:
We could but I'm not sure how we can update the handler context to return sync values and if that's worth it, or do you mean having both promise and sync values being an option for the properties ? For the latter I think having common interface for both routes and connector outweighs the unnecessary promise, wdyt ?